sdk/src/runtime/lib/router.ts

```diff
@@ -167,27 +171,25 @@ export function defineRoutes<TContext = Record<string, any>>(
       }
 
       let { params, handler } = match;
+      routeContext.params = params;
 
-      let handlers = Array.isArray(handler) ? handler : [handler];
+      const handlers = Array.isArray(handler) ? handler : [handler];
       for (const h of handlers) {
         if (isRouteComponent(h)) {
-          const actionResult = await rw.handleAction(ctx);
+          const actionResult = await rw.handleAction(routeContext);
           const serializedEnv = serializeEnv(env);
           const props = { params, env: serializedEnv, ctx };
           return await rw.renderPage({
-            Page: handler as React.FC<Record<string, any>>,
+            Page: h as React.FC<Record<string, any>>,
```

Comment:
Another change I needed to make to get router working again - we're iterating over the handlers so I think we wanted the handler (`h`) itself not the array of handlers here.

Reply:
Ah fuck. That was my bad. Sorry.